### PR TITLE
fix: don't output to api dir, insert xref baseUrl

### LINKED ISSUE #64: docfx: api directory included in xrefmap.yml files
The `xrefmap.yml` files we're generating right now include `api/` in the `href` of every item. We don't want that.

cc @dandhlee @jskeet

### PR DESCRIPTION
**Pull request by tbpg**
Fixes #64.

### CODE REVIEW

**Review by dandhlee — APPROVED** [review]
Looks good! Does the pipeline down the road expect anything to be in `site/api`? Or does it simply grep for `**/*.html`?

File: docpipeline/generate.py [hunk 1]
```diff
@@ -168,29 +163,33 @@ def process_blob(blob, credentials, devsite_template, xrefs):
...
     # The input blob has a "docfx-" prefix; make sure to remove it.
+    xrefmap = site_path.joinpath("xrefmap.yml")
+    xrefmap_lines = xrefmap.read_text().splitlines()
+    # The baseUrl must start with a scheme and domain. With no scheme, docfx
+    # assumes it's a file:// link.
+    base_url = (
+        f"baseUrl: https://cloud.google.com/{metadata.language}/docs/reference/"
+        + f"{metadata.name}/latest/"
+    )
+    # Insert base_url after the YamlMime first line.
```
Comment (jskeet) [inline docpipeline/generate.py:189]: It's possible this is fine - I've always put it *before* the YamlMime line. I realize that's weird, but it's what's always worked for me...

File: docpipeline/generate.py [hunk 2]
```diff
@@ -168,29 +163,33 @@ def process_blob(blob, credentials, devsite_template, xrefs):
...
-        )
-
-    # Copy the xrefmap file to the output directory.
-    xrefmap = site_path.joinpath("xrefmap.yml")
-    shutil.copy(xrefmap, site_api_path)
+        shutil.move(site_path.joinpath("toc.html"), site_path.joinpath("_toc.yaml"))
 
     # Add the prettyprint class to code snippets
-    add_prettyprint(site_api_path)
+    add_prettyprint(site_path)
```
Comment (jskeet) [inline docpipeline/generate.py:171]: I suspect we want to delete manifest.json as well. That's docfx output that I doubt we want to publish (and won't have been publishing before)

Comment (jskeet) [reply docpipeline/generate.py:171]: Note: if with the PR as it is, we *are* uploading manifest.json, but our golden tests aren't failing with unexpected data, we should fix the tests to spot that.

**Review by jskeet — COMMENTED** [review]
Happy with the general idea - a couple of points noted in comments.

File: docpipeline/generate.py (hunk 2 above)
Comment (tbpg) [reply docpipeline/generate.py:171]: Done. I added a test.

We don't have full on golden tests in this repo -- only doc-templates.

File: docpipeline/generate.py (hunk 1 above)
Comment (tbpg) [reply docpipeline/generate.py:189]: They both seem to work. So, I think I'd prefer to put it below the YamlMime.

**Review by tbpg — COMMENTED** [review]
> Does the pipeline down the road expect anything to be in site/api? Or does it simply grep for **/*.html?

No. The HTML is uploaded from the root directory of the HTML.

File: docpipeline/generate.py (hunk 1 above)
Comment (jskeet) [reply docpipeline/generate.py:192]: Fine by me.

**Review by jskeet — APPROVED** [review]

**Review by busunkim96 — APPROVED** [review]

**Pull request merged**